test/UserCohort.spec.js

```diff
@@ -0,0 +1,71 @@
+const { expect } = require("chai");
+const mongoUnit = require("mongo-unit");
+const mongoose = require("mongoose");
+const User = require("User");
+const UserCohort = require("UserCohort");
+const Goal = require("Goal");
+const GoalCohort = require("GoalCohort");
+const Cohort = require("Cohort");
+
+describe("UserCohort", function() {
+  beforeEach(async () => {
+    await mongoUnit.load(require("./fixtures/mongodb/data-default.js"));
+  });
+
+  afterEach(async () => {
+    await mongoUnit.drop();
+  });
+
+  describe("setUserCohort", function() {
+    it("sets user cohort", async () => {
+      const user = await User.findById(
+        mongoose.Types.ObjectId("5dd88892c012321c14267155")
+      );
+      expect(await UserCohort.findForUser(user)).to.not.exist;
+      await UserCohort.setUserCohort(user, "Study Cohort");
+      const userCohort = await UserCohort.findForUser(user);
+      expect(userCohort).to.exist;
+      expect(userCohort).to.be.instanceof(UserCohort);
+      expect(userCohort.cohort).to.eql(
+        mongoose.Types.ObjectId("5ed82fb2a869c32825c74474")
+      );
+    });
+
+    it("gets the user cohort", async () => {
+      const user = await User.findById(
+        mongoose.Types.ObjectId("5dd88892c012321c14267156")
+      );
+      const userCohort = await UserCohort.findForUser(user);
+      expect(userCohort).to.exist;
+      expect(userCohort).to.be.instanceof(UserCohort);
+      expect(userCohort.cohort).to.eql(
+        mongoose.Types.ObjectId("5ed82fb2a869c32825c74474")
+      );
+    });
+
+    it("returns the new user cohort", async () => {
+      const user = await User.findById(
+        mongoose.Types.ObjectId("5dd88892c012321c14267156")
+      );
+      const userCohort = await UserCohort.setUserCohort(
+        user,
+        "New Study Cohort"
+      );
+      const cohort = await Cohort.findOrCreateForName("New Study Cohort");
+      expect(userCohort).to.exist;
+      expect(userCohort).to.be.instanceof(UserCohort);
+      expect(userCohort.cohort).to.eql(cohort.id);
+    });
+
+    it("leaves old cohorts when setting new user cohort", async () => {
```

Comment:
need two cases covered here:

```js
it("removes user from any previously joined GoalCohorts that do not belong to the user's new named Cohort")
it("does NOT remove the user from any previously joined GoalCohorts that do belong to the user's new named Cohort")
```

Reply:
Is the second case even possible? To have joined that cohort, you'd have to be in the named cohort to begin with. And if you're setting your named cohort to something else, then you couldn't have been in that cohort, so you couldn't have joined the other cohort?

Reply:
> Is the second case even possible? To have joined that cohort, you'd have to be in the named cohort to begin with. And if you're setting your named cohort to something else, then you couldn't have been in that cohort, so you couldn't have joined the other cohort?

OK makese sense. Can we guarantee somehow that rejoining the same cohort is a noop?